src/functions/types.ts

```diff
@@ -396,3 +397,40 @@ export type RuntimeUnhandledEventContext<
     // deno-lint-ignore no-explicit-any
     body: any;
   };
+
+/**
+ * @description Used to extract returned args from Async and Sync function handlers
+ */
+type FunctionRuntimeReturnArgs<
+  FunctionReturnType extends
+    | FunctionHandlerReturnArgs<FunctionParameters>
+    | Promise<FunctionHandlerReturnArgs<FunctionParameters>>,
+> = FunctionReturnType extends FunctionHandlerReturnArgs<FunctionParameters>
+  ? FunctionReturnType
+  : FunctionReturnType extends
+    Promise<FunctionHandlerReturnArgs<FunctionParameters>>
+    ? Awaited<FunctionReturnType>
+  : never;
+
+type BaseFunctionRuntimeType<
+  EnrichedFunction extends EnrichedSlackFunctionHandler<
+    FunctionDefinitionArgs<
+      ParameterSetDefinition,
+      ParameterSetDefinition,
+      PossibleParameterKeys<ParameterSetDefinition>,
+      PossibleParameterKeys<ParameterSetDefinition>
+    >
+  >,
+> = {
+  args: Parameters<EnrichedFunction>[number];
+  outputs: FunctionRuntimeReturnArgs<ReturnType<EnrichedFunction>>["outputs"];
+};
+
+/**
+ * @description Used to surface function runtime typescript types from defined functions
+ */
+export type FunctionRuntimeType<
```

Comment:
Food for thought — I read `FunctionRuntimeType` as the type of the function runtime signature. I think we could do better to indicate it's a utility type and what it's doing
1. Pluralize to `FunctionRuntimeTypes` to indicate we're returning multiple types
2. Use a verb to indicate it's a utility that is taking an action rather than responding with the type representation. Suggest `ExtractFunctionRuntimeTypes`

Reply:
I will implement both of these points and apply point `2.` to `RuntimeType` -> `ExtractRuntimeType` because using a verb had crossed my mind in the past and for consistency purposes